### PR TITLE
feat[WIP]: Cart

### PR DESCRIPTION
**Pull request by saadahmadhere**
This PR mainly focuses on:
1. Add to cart functionality.
2. Remove from cart functionality.
3. Change the number of items in cart.
4. Calculating the total price of items present in the Cart.

### CODE REVIEW

File: src/Components/CardHorizontal/CardHorizontal.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import "./CardHorizontal.css";
+
+const CardHorizontal = ({ product }) => {
+  const {
+    state: { cart },
+    dispatch,
+  } = useData();
+
+  const itemQuantityHandler = (e, product) => {
+    console.log(e.target.value, product);
```
Comment (abhij1607) [inline src/Components/CardHorizontal/CardHorizontal.jsx]: cleanup console.log

File: src/Components/CardHorizontal/CardHorizontal.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    dispatch({
+      type: "CHANGE_ITEM_QUANTITY",
+      payload: { id: product.id, qty: e.target.value },
+    });
+  };
+
+  const removeCartItemHandler = (_id) => {
+    dispatch({ type: "REMOVE_FROM_CART", payload: _id });
+    console.log(_id);
```
Comment (abhij1607) [inline src/Components/CardHorizontal/CardHorizontal.jsx]: cleanup console.log

File: src/Components/CardHorizontal/CardHorizontal.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  };
+  console.log(cart);
```
Comment (abhij1607) [inline src/Components/CardHorizontal/CardHorizontal.jsx]: cleanup console.log

File: src/Screens/CartScreen/CartScreen.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { CheckOutDetails } from "../../Components";
+import { EmptyCart } from "../";
+
+const CartScreen = () => {
+  const {
+    state: { cart },
+    dispatch,
+  } = useData();
+
+  console.log(cart);
```
Comment (abhij1607) [inline src/Screens/CartScreen/CartScreen.jsx]: cleanup console.log

**Review by abhij1607 — COMMENTED** [review]
Overall clean code | | Nice work with the EmptyCart component

It is good to clean up Console Log, so pointed them in the comments

File: src/Components/Card-ProductPage/Card.jsx [hunk 5]
```diff
@@ -1,6 +1,21 @@
...
+import { useData } from "../../Helper";
+import { Link } from "react-router-dom";
 
 const Card = ({ product }) => {
+  const {
+    state: { cart },
+    dispatch,
+  } = useData();
+
+  console.log(cart);
```
Comment (junaidshaikh-js) [inline src/Components/Card-ProductPage/Card.jsx]: console logs should be removed.

File: src/Screens/CartScreen/CartScreen.jsx (hunk 4 above)
Comment (saadahmadhere) [reply src/Screens/CartScreen/CartScreen.jsx]: done

File: src/Components/CardHorizontal/CardHorizontal.jsx (hunk 3 above)
Comment (saadahmadhere) [reply src/Components/CardHorizontal/CardHorizontal.jsx]: done

File: src/Components/CardHorizontal/CardHorizontal.jsx (hunk 2 above)
Comment (saadahmadhere) [reply src/Components/CardHorizontal/CardHorizontal.jsx]: done

File: src/Components/CardHorizontal/CardHorizontal.jsx (hunk 1 above)
Comment (saadahmadhere) [reply src/Components/CardHorizontal/CardHorizontal.jsx]: done

**Review by junaidshaikh-js — APPROVED** [review]
Great work! 

Good folder structure. Clean code. Can be merged.

File: src/Components/Card-ProductPage/Card.jsx (hunk 5 above)
Comment (saadahmadhere) [reply src/Components/Card-ProductPage/Card.jsx]: done

**Pull request merged**